Sage/carbon
# fix(tabs, icon): override behaviour of icon with tooltip when rendered inside tab title

fix #4663

### Proposed behaviour


When an `Icon` with a `Tooltip` is rendered within a `TabTitle`, the show/ hide functionality is
overridden so that the `Tooltip` displays when the `TabTitle` has focus or the mouse hovers over and
is hidden when loses focus or the mouse leaves. The `tabIndex` for the `Icon` is also set to undefined
so that it cannot be tabbed to.

![fixed tabs](https://user-images.githubusercontent.com/44157880/187725690-f9ff5d9c-64c2-476d-99cd-5c96a106084e.gif)

### Current behaviour


`Icon`s with `Tooltip` inside `TabTitle` can be tabbed to which breaks the keyboard navigation for the `Tabs` and the content within them.

![broken tabs](https://user-images.githubusercontent.com/44157880/187725303-63681e1c-71ad-4751-94f9-9f32cab0ad0c.gif)

### Checklist




#### QA


### Additional context


Any improvements to voiceover will be done in a separate ticket as will require consulting wit UX to decide what expected behaviour should be.

### Testing instructions



The following CodeSandbox is an example of the broken behaviour.
You can see the new behaviour by looking at the version in the comment by `codesandbox[bot]`.


The linked issue is missing design-tokens dependency and has not implemented the `CarbonProvider`, QA should use the one built from the sandbox below when testing 

https://codesandbox.io/s/tender-payne-2iqv8x?file=/src/index.js
